src/solvers/thermal_solver.cpp

```diff
@@ -6,32 +6,32 @@
 
 #include "thermal_solver.hpp"
 
+const int num_fields = 1;
```

Comment:
Minor nitpick, please use "constexpr int" instead of "const int" since constexpr's are evaluated at compile time not run time when you do hard-coded "magic" numbers.

Reply:
[constexpr ALL the things!](https://www.youtube.com/watch?v=PJwd4JLYJJY). Addressed in next PR.